lib/helpers/deployments.rb

```diff
@@ -1,17 +1,23 @@
+require 'plugins/mcollective'
+
 module Deployments # rubocop:disable Style/Documentation
   def deploy(branch, deleted)
-    if settings.use_mco_ruby
-      result = mco(branch).first
+    # Currently requires puppet_webhook to be run as a non-root
+    # user with access to running the MCollective client.
+    if settings.use_mcollective
+      result = PuppetWebhook::Mcollective.new('r10k',
+                                              'deploy',
+                                              {
+                                                dtimeoute: settings.discovery_timeout,
+                                                timeout: settings.client_timeout
+                                              },
+                                              settings.client_cfg,
+                                              environment: branch).run.first
       raise result.results[:statusmsg] unless result.results[:statuscode].zero?
 
       message = result.results[:statusmsg]
     else
-      command = if settings.use_mcollective
-                  "#{settings.command_prefix} mco r10k deploy #{branch} #{settings.mco_arguments}"
-                else
-                  # If you don't use mcollective then this hook needs to be running as r10k's user i.e. root
-                  "#{settings.command_prefix} r10k deploy environment #{branch} #{settings.r10k_deploy_arguments}"
-                end
+      command = "#{settings.command_prefix} r10k deploy environment #{branch} #{settings.r10k_deploy_arguments}"
```

Comment:
Are all these variables sanitized already? eg, what if `branch` was set to `; rm -rf /;`?

Reply:
We are not sanitizing that, no. I'm not sure if that config parameter will still be around in the future. I'm planning on (attempting) to replace r10k open3 calls with the r10k libs (if possible).

TL;DR - I want to remove `run_command` if at all possible.